packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -362,6 +363,21 @@ export abstract class AbstractMessageManager<
     this.updateMessage(message, false);
   }
 
+  /**
+   * Sets the message defer property.
+   *
+   * @param messageId - The id of the Message to update.
+   * @param deferAsSigned - The defer value to add in the message params.
+   */
+  setDeferSetAsSigned(messageId: string, deferAsSigned: any) {
```

Comment:
I don't follow why we need to do this after creating the message.

The standard entry points are `newUnsignedMessage`, `newUnsignedPersonalMessage`, and `newUnsignedTypesMessage`. All of three of these use types extended `AbstractMessageParams` which includes the `deferAsSigned` boolean already.

Reply:
I just mean why set it after the fact with an additional method given when the message is created, we're already in an MMI context and know we'll be using a custodian to do the signing.

We could instead inject the property before it reaches the controller by updating the middleware bindings below to use a wrapper function in which we force the params to set `deferSetSign` using an MMI code fence.

https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L1488